apps/tangle-dapp/containers/UpdateNominationsTxContainer/UpdateNominationsTxContainer.tsx

```diff
@@ -29,6 +34,24 @@ const UpdateNominationsTxContainer: FC<UpdateNominationsTxContainerProps> = ({
     [getActiveValidators.name],
     ([, ...args]) => getActiveValidators(...args)
   );
+  const { data: waitingValidatorsData } = useSWR(
+    [getWaitingValidators.name],
+    ([, ...args]) => getWaitingValidators(...args)
+  );
+
+  const [maxNominationQuota, setMaxNominationQuota] = useState<number>(0);
+
+  useEffect(() => {
+    getMaxNominationQuota().then((maxNominationQuota) => {
+      setMaxNominationQuota(maxNominationQuota ? maxNominationQuota : 16);
+    });
+  }, []);
+
+  const allValidators = useMemo(() => {
+    if (!activeValidatorsData || !waitingValidatorsData) return [];
+
+    return [...activeValidatorsData, ...waitingValidatorsData];
+  }, [activeValidatorsData, waitingValidatorsData]);
```

Comment:
Could you please deduplicate this code? You can achieve this by creating a custom hook to encapsulate the logic. Afterwards, ensure to reuse the hook in both this file and `apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx`.